lib/elixir/lib/file/stream.ex

```diff
@@ -73,7 +73,8 @@ defmodule File.Stream do
       start_fun =
         fn ->
           case :file.open(path, read_modes(modes)) do
-            {:ok, device}    -> device
+            {:ok, device}    ->
+              if :strip_bom in modes, do: strip_bom(device), else: device
```

Comment:
I think this is counter-intuitive use of guard. What do you think about something like this:
```
case :file.open(path, read_modes(modes)) do
  {:ok, device} ->
    device
  {:error, reason} ->
    ...
end
```
and
```
def strip_bom(device, modes) when :strip_bom in modes, do: ..
def strip_bom(device, _modes), do: device
```